airflow_clickhouse_plugin/hooks/clickhouse_hook.py

```diff
@@ -19,19 +19,29 @@ def __init__(
 
     def get_conn(self) -> Client:
         conn = self.get_connection(self.clickhouse_conn_id)
+        extra_connection_kwargs = {}
+        if conn.extra_dejson.get('client_name'):
+            extra_connection_kwargs['client_name'] = conn.extra_dejson['client_name']
```

Comment:
I guess it could be `None`, e.g. `"client_name": null` can be specified in extra parameters if needed. So you should better use `if 'client_name' in conn.extra_dejson` condition.

Reply:
It can't be `None` otherwise next exception will be thrown:
`TypeError: can only concatenate str (not "NoneType") to str`
in [this ](https://github.com/mymarilyn/clickhouse-driver/blob/629d1e1ea0684b806c1526a332d8cb5de517e081/clickhouse_driver/connection.py#L129) place

Reply:
`"client_name"` is not a representative example, you are right. I was talking about `null` values in general: maybe `connect_timeout` or `verify_cert` could be possibly `None`. Or any other future arguments will accept `None` as a value.

Reply:
I understand what is the point, but using a _dynamic_ approach, in addition, you do not need to check each parameter?

Reply:
By dynamic approach I mean that we do not need to hardcode a set of arguments we are using. We completely rely on clickhouse-driver and push all the arguments down to `Connection.__init__`.